pkg/apis/middleware/scope.go

```diff
@@ -35,6 +35,10 @@ type RequestScope struct {
 	// ClearSession indicates whether the user should be logged out or not.
 	ClearSession bool
 
+	// IntrospectToken indicates whether to introspect the session token.
+	// This is set if the request has the header `X-Oauth2-Proxy-Introspect-Token`.
```

Comment:
Is this approach wise? Shouldn't the introspection always happen? This way the client decides when to redo the introspection, right?

Reply:
Unless the client is Nginx, my use case here is to choose what to introspect. I haven't given much thought about cli options with this flag, maybe add `--introspection-enable-header` which enables this behavior.

E.g. if you want to introspect all write queries, but not read. Or just /admin.

Reply:
I added the flag, make sense!

I guess it should however also enable the sessiontoken creater.

Reply:
I made it such that my header introspection updates the session.

Could we enable that always for --introspect-token and skip the createSessionFromIntrospectedToken?

Reply:
I made a patch doing that. It's wip though, needs testing obviously.